src/Package/Impl/DataInspect/Viewers/DebugObjectEvaluator.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Microsoft.R.Debugger;
+using Microsoft.R.Host.Client;
+using Microsoft.VisualStudio.R.Package.Repl;
+
+namespace Microsoft.VisualStudio.R.Package.DataInspect.Viewers {
+    [Export(typeof(IDataObjectEvaluator))]
+    public sealed class DebugObjectEvaluator : IDataObjectEvaluator {
+        private readonly IDebugSessionProvider _debugSessionProvider;
+        private readonly IRSessionProvider _rSessionProvider;
+        private DebugSession _debugSession;
+        private IRSession _rSession;
+
+        [ImportingConstructor]
+        public DebugObjectEvaluator(IRSessionProvider rSessionProvider, IDebugSessionProvider debugSessionProvider) {
+            _rSessionProvider = rSessionProvider;
+            _debugSessionProvider = debugSessionProvider;
+        }
+
+        public async Task<DebugEvaluationResult> EvaluateAsync(string expression, DebugEvaluationResultFields fields) {
+            await TaskUtilities.SwitchToBackgroundThread();
+
+            var debugSession = await GetDebugSessionAsync();
+            var frames = await debugSession.GetStackFramesAsync();
+
+            return await frames.Last().EvaluateAsync(expression, fields) as DebugValueEvaluationResult;
```

Comment:
On the oft chance that stack frame list is empty (I think this should never happen, but I'm not 100% sure), can you check and throw (or return null, whichever is more appropriate) here?